environment/src/main/java/com/sequenceiq/environment/environment/service/EnvironmentService.java

```diff
@@ -207,9 +208,25 @@ public CreateAWSEnvironmentRequest getCreateAWSEnvironmentForCli(EnvironmentDto
         return environmentRepository.findByResourceCrnAndAccountIdAndArchivedIsFalse(resourceCrn, accountId);
     }
 
+    /**
+     * The CIDR could not be edited so the first step we clear the the existed CIDR every time.
+     * We are assuming the security access has the proper security group ids and we don't check it again.
+     * If the knox or default security groups are blank we don't set
+     * @param environment the security access will be update for this environment
+     * @param securityAccess this should contains the knox and default security groups
+     */
+    void editSecurityAccess(Environment environment, SecurityAccessDto securityAccess) {
+        environment.setCidr(null);
+        if (StringUtils.isNotBlank(securityAccess.getDefaultSecurityGroupId())) {
+            environment.setDefaultSecurityGroupId(securityAccess.getDefaultSecurityGroupId());
+        }
+        if (StringUtils.isNotBlank(securityAccess.getSecurityGroupIdForKnox())) {
+            environment.setSecurityGroupIdForKnox(securityAccess.getSecurityGroupIdForKnox());
+        }
+    }
+
     void setSecurityAccess(Environment environment, SecurityAccessDto securityAccess) {
         if (securityAccess != null) {
-            environment.setCidr(securityAccess.getCidr());
```

Comment:
this is not good as EnvironmentCreationService is also using this method and CIDR setting is impossible with this change

Reply:
hm, I revert this